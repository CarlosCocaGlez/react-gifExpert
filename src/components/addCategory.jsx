import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {
    const [ inputValue, setInputValue ] = useState('')

    const onInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const onSubmit = (e) => {
        e.preventDefault();
        const newInputValue = inputValue.trim();
        if (newInputValue.length <= 1) return;
        // addCategory((category) => [inputValue, ...category])
        onNewCategory( newInputValue )
        setInputValue('')
    }

    return (
        <form onSubmit={ onSubmit }>
            <input
                type='text'
                placeholder='Buscar gifs'
                onChange={ onInputChange }
                value={ inputValue }
            />
        </form>
    )
}