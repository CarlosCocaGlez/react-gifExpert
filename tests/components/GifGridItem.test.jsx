import {render, screen} from "@testing-library/react";
import {GifGridItem} from "../../src/components/GifGridItem";

describe('Pruebas en GifGridItem', () => {
    const title = 'Luffy';
    const url = 'https://one-piece.com/luffy.jpg'
    test('Debe hacer match con el snpashot', () => {
        const { container } = render( <GifGridItem url={ url } title={ title } />)
        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifGridItem url={ url } title={ title } />)
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    });

    test('Debe de mostrar el titulo en el componente', () => {
        render(<GifGridItem url={ url } title={ title } />);
        expect(screen.getByText( title )).toBeTruthy();
    })
});