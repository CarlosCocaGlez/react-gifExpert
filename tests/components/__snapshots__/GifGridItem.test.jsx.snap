// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe hacer match con el snpashot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Luffy"
      src="https://one-piece.com/luffy.jpg"
    />
    <p>
      Luffy
    </p>
  </div>
</div>
`;
